pkg/cosigner/image_signer.go

```diff
@@ -0,0 +1,91 @@
+package cosigner
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/BurntSushi/toml"
+	"github.com/buildpacks/lifecycle"
+	"github.com/sigstore/cosign/cmd/cosign/cli"
+)
+
+type ImageSigner struct {
+	Logger *log.Logger
+}
+
+var (
+	cliSignCmd     = cli.SignCmd
+	secretLocation = "/var/build-secrets"
+)
+
+func NewImageSigner(logger *log.Logger) *ImageSigner {
+	return &ImageSigner{
+		Logger: logger,
+	}
+}
+
+func (s *ImageSigner) Sign(reportFilePath string, annotations map[string]interface{}) error {
+	var report lifecycle.ExportReport
+	_, err := toml.DecodeFile(reportFilePath, &report)
+	if err != nil {
+		return fmt.Errorf("toml decode: %v", err)
+	}
+
+	if len(report.Image.Tags) < 1 {
+		s.Logger.Println("no image tag to sign")
+		return nil
+	}
+
+	cosignFolders := findCosignFolders(secretLocation)
+	if len(cosignFolders) == 0 {
+		s.Logger.Println("no keys found for cosign signing")
+		return nil
+	}
+
+	refImage := report.Image.Tags[0]
+
+	ctx := context.Background()
+	for _, cosignFolder := range cosignFolders {
+		cosignKeyFile := fmt.Sprintf("%s/cosign.key", cosignFolder)
+		cosignPasswordFile := fmt.Sprintf("%s/cosign.password", cosignFolder)
+
+		ko := cli.KeyOpts{KeyRef: cosignKeyFile, PassFunc: func(bool) ([]byte, error) {
```

Comment:
It is unfortunate that cosign does not seem to support providing an alternative keychain. It works to rely on the the default keychain here but, when we want to sign builder images we will need to provide the ggcr keychain to cosign.

Reply:
Might be worth raising on issue on: https://github.com/sigstore/cosign/issues

Reply:
https://github.com/sigstore/cosign/issues/666

Reply:
Oh, they made it the day after my issue was raised :) Perfect! 
Thanks for letting me know